aten/src/ATen/native/Loss.cpp

```diff
@@ -390,20 +390,22 @@ Tensor& mse_loss_backward_out(Tensor& grad_input, const Tensor& grad_output,
 }
 
 Tensor l1_loss(const Tensor& input, const Tensor& target, int64_t reduction) {
-  auto loss = input.sub(target).abs_();
+  auto diff = input.sub(target);
+  auto loss = diff.is_complex() ? diff.abs() : diff.abs_();
   return apply_loss_reduction(loss, reduction);
 }
 
 Tensor& l1_loss_out(Tensor&result, const Tensor& input, const Tensor& target, int64_t reduction) {
+  auto diff = input.sub(target);
```

Comment:
This is interesting. In the previous code `input - target` is out of place if Reduction::None, but inplace otherwise. With this change it will always be out of place. 

I'm not sure why that distinction existed previously, but it seems like this change can preserve the inplace subtraction in the `else` clause?